databroker/core.py

```diff
@@ -535,6 +535,82 @@ def documents_to_xarray(*, start_doc, stop_doc, descriptor_docs,
     return xarray.merge(datasets)
 
 
+def canonical(*, start, stop, entries, fill, strict_order=True):
+    """
+    Yields documents from this Run in chronological order.
+
+    Parameters
+    ----------
+    start_doc : dict
+        RunStart Document
+    stop_doc : dict
+        RunStop Document
+    entries : dict
+        A dict of the BlueskyRun's entries.
+    fill: {'yes', 'no'}
+        If fill is 'yes', any external data referenced by Event documents
+        will be filled in (e.g. images as numpy arrays). This is typically
+        the desired option for *using* the data.
+        If fill is 'no', the Event documents will contain foreign keys as
+        placeholders for the data. This option is useful for exporting
+        copies of the documents.
+    strict_order : bool, optional
+        documents are strictly yielded in ascending time order.
+    """
+    run_start_uid = start['uid']
+    history = set()
+
+    # Special case for 'delayed' since it *is* supported in the local mode
```

Comment:
The value `'delayed'` should be allowed in the local-access case. Note the difference between the `canonical` method on remote and the `canonical` method on local in this respect. We could fix this by moving _just_ the `fill` validation step into the two separate methods.

Reply:
fixed